### PR TITLE
4614: Enable call off party task

### PR DESCRIPTION
**Pull request by kat1906**
Activity: [AB#4614](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/4614)
Task: [AB#6966](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/6966)

### CODE REVIEW

File: app/pages/task-list/existingorder/ui.test.js [hunk 1]
```diff
@@ -145,6 +145,26 @@ test('should render the first task and tag it as complete', async (t) => {
...
+  await t.debug()
+    .expect(taskList.exists).ok()
+    .expect(secondTask.exists).ok()
+    .expect(await extractInnerText(secondTask.find('h2 span'))).eql('2.')
+    .expect(await extractInnerText(secondTask.find('h2 div'))).eql('Organisation information')
+    .expect(secondTaskFirstItem.exists).ok()
+    .expect(await extractInnerText(secondTaskFirstItem)).eql('Provide Call-off Ordering Party information')
+    .expect(secondTaskFirstItem.find('a').exists).ok()
+    .click(secondTaskFirstItem.find('a'))
+    .expect(getLocation()).eql(`http://localhost:1234${baseUrl}/organisation/order-id/call-off-ordering-party`);
```
Comment (alibombaye) [inline app/pages/task-list/existingorder/ui.test.js:165]: sweet!

File: app/pages/task-list/helpers/generateTaskList.test.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+  },
+  {
+    id: 'task2item2',
+    status: 'incomplete',
+  },
+];
+// [
+//   { taskName: 'task 1', items: [ [Object] ] },
+//   { taskName: 'task 2', items: [ [Object], [Object] ] }
+// ]
```
Comment (alibombaye) [inline app/pages/task-list/helpers/generateTaskList.test.js:44]: comments???

File: app/pages/task-list/helpers/generateTaskList.test.js [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+describe('generateTaskList', () => {
+  describe('taskName', () => {
+    it('should return task.name as taskName in tasks array for each task in manifest', () => {
```
Comment (alibombaye) [inline app/pages/task-list/helpers/generateTaskList.test.js:47]: I find this test description a little confusing.  Can you change it please

File: app/pages/task-list/helpers/generateTaskList.test.js [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+    });
+  });
+
+  describe('items', () => {
+    it('should return an items array with the same number of items as section in the manifest for each task', () => {
+      const taskList = generateTaskList({ orderId, taskListManifest });
+      taskList.forEach((task, i) => {
+        expect(task.items.length).toEqual(taskListManifest.tasks[i].sections.length);
+      });
+    });
```
Comment (alibombaye) [inline app/pages/task-list/helpers/generateTaskList.test.js:61]: 👍  nice

File: app/pages/task-list/helpers/generateTaskList.test.js (hunk 2 above)
Comment (kat1906) [reply app/pages/task-list/helpers/generateTaskList.test.js:44]: 😬 soz

File: app/pages/task-list/helpers/generateTaskList.test.js (hunk 3 above)
Comment (kat1906) [reply app/pages/task-list/helpers/generateTaskList.test.js:47]: yeah me too 😄

File: app/pages/task-list/helpers/generateTaskList.test.js [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+      const taskList = generateTaskList({ orderId, taskListManifest });
+      taskList.forEach((task, i) => {
+        task.items.forEach((item, j) => {
+          expect(item.description)
+            .toEqual(taskListManifest.tasks[i].sections[j].title);
+        });
+      });
+    });
+
+    it('should not add href to item if no section data is available if the item has dependencies', () => {
```
Comment (alibombaye) [inline app/pages/task-list/helpers/generateTaskList.test.js:70]: Just stick an and in there so it reads little better
should not add href to item if no section data is available `and` if the item has dependencies

File: app/pages/task-list/helpers/generateTaskList.test.js [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+      taskList.forEach((task, i) => {
+        task.items.forEach((item, j) => {
+          if (item.href) {
+            expect(item.href)
+              .toEqual(`${baseUrl}/organisation/${orderId}/${taskListManifest.tasks[i].sections[j].id}`);
+          }
+        });
+      });
+    });
+
```
Comment (alibombaye) [inline app/pages/task-list/helpers/generateTaskList.test.js:99]: Do we want a few more tests for when we do have sectionData passed in and that sectionData has one of dependencies as complete then that item is enabled.

Also if the sectionData passed in that data has a dependency as incomplete then that item is not enabled..

Perhaps describe blocks for with and without sectionData provided

File: app/pages/task-list/helpers/generateTaskList.test.js (hunk 3 above)
Comment (alibombaye) [reply app/pages/task-list/helpers/generateTaskList.test.js]: I didn't even know what to suggest here 😂

File: app/pages/task-list/helpers/generateTaskList.test.js (hunk 6 above)
Comment (kat1906) [reply app/pages/task-list/helpers/generateTaskList.test.js:99]: ok I will work on those - I think I can add more for the href too

Comment (alibombaye) [reply app/pages/task-list/helpers/generateTaskList.test.js:99]: 👍 besides that it all looks great. Think we have a nice little scalable solution here. Just testing it later on is going to be the bulk of the work

Comment (kat1906) [reply app/pages/task-list/helpers/generateTaskList.test.js:99]: ok you meant href 😄

Comment (alibombaye) [reply app/pages/task-list/helpers/generateTaskList.test.js:84]: haha yes

**Pull request merged**